### PR TITLE
Add support for flexible edge routing

### LINKED ISSUE #60: Support flexible edge routing [8]
Currently, the edges are automatically routed, which is fine. However, if the user changes the routing, the new routing is not persisted and gets lost on the next update:
Peek 2019-11-14 09-11

### PR DESCRIPTION
**Pull request by sgraband**
Resolves #60

Signed-off-by: simonGraband <simon.graband@tum.de>

### CODE REVIEW

File: server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreFacade.java [hunk 1]
```diff
@@ -206,7 +207,9 @@ public void updateShape(Shape shape, GShapeElement shapeElement) {
 	public void updateEdge(Edge edge, GEdge gEdge) {
 		edge.getBendPoints().clear();
 		if (gEdge.getRoutingPoints() != null) {
-			edge.getBendPoints().addAll(gEdge.getRoutingPoints());
+			ArrayList<GPoint> gPoints = new ArrayList<>();
+            gEdge.getRoutingPoints().forEach(p -> gPoints.add(GraphUtil.copy(p)));
```
Comment (eneufeld) [inline server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreFacade.java]: formatting?

File: server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreServerLauncher.java [hunk 2]
```diff
@@ -26,7 +28,7 @@ public static void main(String[] args) {
 		EcorePackage.eINSTANCE.eClass();
 		ElkLayoutEngine.initialize(new LayeredMetaDataProvider());
 		BasicConfigurator.configure();
-		
+		Logger.getRootLogger().setLevel(Level.DEBUG);
```
Comment (eneufeld) [inline server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreServerLauncher.java:31]: do we need this?

File: server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/gmodel/ClassifierNodeFactory.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ public GNode create(EClass eClass) {
 				.layout(GConstants.Layout.VBOX) //
 				.addCssClass(CSS.NODE) //
 				.add(buildHeader(eClass))//
-				.add(createLabeledChildrenCompartment(eClass.getEAttributes()));
+				.add(createLabeledChildrenCompartment(eClass.getEAttributes(),eClass));
```
Comment (eneufeld) [inline server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/gmodel/ClassifierNodeFactory.java]: formatting?

File: server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/gmodel/ClassifierNodeFactory.java [hunk 4]
```diff
@@ -83,7 +83,7 @@ public GNode create(EEnum eEnum) {
 				.layoutOptions(new GLayoutOptionsBuilder().resizeContainer(true).build()) //
 				.addCssClass(CSS.NODE) //
 				.add(buildHeader(eEnum))//
-				.add(createLabeledChildrenCompartment(eEnum.getELiterals()));
+				.add(createLabeledChildrenCompartment(eEnum.getELiterals(),eEnum));
```
Comment (eneufeld) [inline server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/gmodel/ClassifierNodeFactory.java]: formatting

**Review by eneufeld — COMMENTED** [review]
Looks good, could you please recheck the formatting?

File: server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreServerLauncher.java (hunk 2 above)
Comment (sgraband) [reply server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreServerLauncher.java:31]: This was added by @tortmayr.

Comment (tortmayr) [reply server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreServerLauncher.java:31]: Most of the glsp logging on server side is set to the `Debug` Level but `BasicConfiguration.configure()` sets the Log level to `Info` which hides lots of usefull log messages

Comment (eneufeld) [reply server/ecore-glsp/src/main/java/com/eclipsesource/glsp/ecore/EcoreServerLauncher.java:31]: Thank you for clarifying

**Review by eneufeld — APPROVED** [review]
I think the formatting is still a bit off, but we can fix this later

**Pull request merged**